streamer/src/nonblocking/recvmmsg.rs

```diff
@@ -16,7 +13,7 @@ pub async fn recv_mmsg(
     packets: &mut [Packet],
 ) -> io::Result</*num packets:*/ usize> {
     debug_assert!(packets.iter().all(|pkt| pkt.meta() == &Meta::default()));
-    let count = cmp::min(NUM_RCVMMSGS, packets.len());
+    let count = packets.len();
```

Comment:
why is it safe to get rid of the `min` here?

Reply:
The callers of this function are never making batches bigger than this constant. Having this check is redundant.

If you really insist, I could turn it into `debug_assert!`, to catch some potential mistake of calling this function with too big batches. But doing that comparison in production on every packet receive. There is a slight chance that LLVM passes optimize it out anyway. :joy: 

Also, I removed the `NUM_RCVMMSG` constant, because it just repeats `PACKETS_PER_BATCH`.

I can propose that as a separate PR, to remove the diff here.

Reply:
Ok quick check implies this function is never actually used in agave outside of tests. I'll go plant a DCOU on it.

Reply:
The same code is in other places though. I'm gonna clean it up in separate PRs.

I've already submitted one for removing the reduntant `NUM_RCMMMSGS` constant: #6955